# HG changeset patch
# User David Bateman <dbateman@free.fr>
# Date 1207759206 14400
#      Wed Apr 09 12:40:06 2008 -0400
# Node ID e44e4cd2129d83e4dfff2072679c7687a5a65648
# Parent  80b78635eefe98666f52976c695b98b450a7541e
Don't allow the use of lists in assert.m

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-04-09  David Bateman  <dbateman@free.fr>
+
+	* testfun/assert.m: Don't allow cond and expected to be lists.
+
 2008-04-04  John W. Eaton  <jwe@octave.org>
 
 	* strings/Makefile.in (SOURCES): Add isstrprop.m to the list.
 
 	* Makefile.in, audio/Makefile.in, control/Makefile.in,
 	* Makefile.incontrol/base/Makefile.in, control/hinf/Makefile.in,
 	* Makefile.incontrol/obsolete/Makefile.in,
 	* Makefile.incontrol/system/Makefile.in, control/util/Makefile.in,
diff --git a/scripts/testfun/assert.m b/scripts/testfun/assert.m
--- a/scripts/testfun/assert.m
+++ b/scripts/testfun/assert.m
@@ -72,29 +72,16 @@ function assert (cond, expected, tol)
 
   coda = "";
   iserror = 0;
   if (nargin == 1)
     if (! isnumeric (cond) || ! all (cond(:)))
       error ("assert %s failed", in); # say which elements failed?
     endif
   
-  elseif (is_list (cond))
-    if (! is_list (expected) || length (cond) != length (expected))
-      iserror = 1;
-    else
-      try
-	for i = 1:length (cond)
-	  assert (nth (cond, i), nth (expected, i));
-	endfor
-      catch
-	iserror = 1;
-      end_try_catch
-    endif
-
   elseif (ischar (expected))
     iserror = (! ischar (cond) || ! strcmp (cond, expected));
 
   elseif (iscell (expected))
     if (! iscell (cond) || any (size (cond) != size (expected)))
       iserror = 1;
     else
       try
